store/mockstore/unistore/rpc.go

```diff
@@ -152,6 +152,25 @@ func (c *RPCClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.R
 	case tikvrpc.CmdTxnHeartBeat:
 		resp.Resp, err = c.usSvr.KvTxnHeartBeat(ctx, req.TxnHeartBeat())
 	case tikvrpc.CmdBatchGet:
+		batchGetReq := req.BatchGet()
+		failpoint.Inject("rpcBatchGetResult", func(val failpoint.Value) {
+			switch val.(string) {
+			case "keyError":
+				failpoint.Return(&tikvrpc.Response{
+					Resp: &kvrpcpb.BatchGetResponse{Error: &kvrpcpb.KeyError{
+						Locked: &kvrpcpb.LockInfo{
+							PrimaryLock: batchGetReq.Keys[0],
+							LockVersion: batchGetReq.Version - 1,
+							Key:         batchGetReq.Keys[0],
+							LockTtl:     2000,
+							TxnSize:     1,
+							LockType:    kvrpcpb.Op_Put,
+						},
+					}},
+				}, nil)
+			}
+		})
+
 		resp.Resp, err = c.usSvr.KvBatchGet(ctx, req.BatchGet())
```

Comment:
Consider use `batchGetReq` instead of `req.BatchGet()` to make the purpose more clear. It doesn't matter though.

Reply:
Fixed. Could you approve the PR again?